core/trino-main/src/main/java/io/trino/execution/scheduler/NodeAssignmentStats.java

```diff
@@ -17,6 +17,7 @@
 import io.trino.execution.PartitionedSplitsInfo;
```

Comment:
This PR seems pretty complex. Why can't we just have a future in `io.trino.metadata.StaticCatalogStore#loadCatalogs` that would be set once all catalogs are loaded?

Reply:
Not with the current implementation of `DiscoveryNodeManager`. 

DiscoveryNodeManager listens on an airlift `ServiceSelector` and populates the list of active catalogs by the `catalogHandleIds` property on the service. This property is populated in the main `Server.java`.

Given catalogs are loaded statically, is this complexity of an airlift `ServiceSelector` needed?

Reply:
I will defer review to @dain on this one